extras/indexing/src/main/java/mvm/rya/indexing/pcj/matching/PCJOptimizer.java

```diff
@@ -76,273 +80,272 @@
+                matcher.matchPCJ(pcj);
+            }
+
+            node.replaceWith(matcher.getQuery());
+            Set<TupleExpr> unmatched = matcher.getUnmatchedArgs();
+            PCJOptimizerUtilities.relocateFilters(matcher.getFilters());
+
+            for (TupleExpr tupleExpr : unmatched) {
+                tupleExpr.visit(this);
+            }
+        }
+
+        @Override
+        public void meet(LeftJoin node) {
+            PCJMatcher matcher = PCJMatcherFactory.getPCJMatcher(node);
+            for (ExternalTupleSet pcj : pcjs) {
+                matcher.matchPCJ(pcj);
+            }
+
+            node.replaceWith(matcher.getQuery());
+            Set<TupleExpr> unmatched = matcher.getUnmatchedArgs();
+            PCJOptimizerUtilities.relocateFilters(matcher.getFilters());
+
+            for (TupleExpr tupleExpr : unmatched) {
+                tupleExpr.visit(this);
+            }
+        }
+
+        @Override
+        public void meet(Filter node) {
+            PCJMatcher matcher = PCJMatcherFactory.getPCJMatcher(node);
+            for (ExternalTupleSet pcj : pcjs) {
+                matcher.matchPCJ(pcj);
+            }
+
+            node.replaceWith(matcher.getQuery());
+            Set<TupleExpr> unmatched = matcher.getUnmatchedArgs();
+            PCJOptimizerUtilities.relocateFilters(matcher.getFilters());
+
+            for (TupleExpr tupleExpr : unmatched) {
+                tupleExpr.visit(this);
+            }
+        }
+
+    }
+
+    /**
+     *
+     *
+     * @param conf
+     *            - client configuration
+     *
+     * @return - list of {@link ExternalTupleSet}s or PCJs that are either
+     *         specified by user in Configuration or exist in system.
+     *
+     * @throws MalformedQueryException
+     * @throws SailException
+     * @throws QueryEvaluationException
+     * @throws TableNotFoundException
+     * @throws AccumuloException
+     * @throws AccumuloSecurityException
+     * @throws PcjException
+     */
+    private static List<ExternalTupleSet> getAccIndices(Configuration conf)
+            throws MalformedQueryException, SailException,
+            QueryEvaluationException, TableNotFoundException,
+            AccumuloException, AccumuloSecurityException, PcjException {
+
+        requireNonNull(conf);
+        String tablePrefix = requireNonNull(conf
+                .get(RdfCloudTripleStoreConfiguration.CONF_TBL_PREFIX));
+        Connector c = requireNonNull(ConfigUtils.getConnector(conf));
+        List<String> tables = null;
+
+        if (conf instanceof RdfCloudTripleStoreConfiguration) {
+            tables = ((RdfCloudTripleStoreConfiguration) conf).getPcjTables();
+        }
+                ;
+        Map<String, String> indexTables = Maps.newLinkedHashMap();
+        PrecomputedJoinStorage storage = new AccumuloPcjStorage(c, tablePrefix);
+        PcjTableNameFactory pcjFactory = new PcjTableNameFactory();
+
+        if (tables != null && !tables.isEmpty()) {
```

Comment:
Style: This could be made more human readable if each condition was a variable. Example:

boolean tablesConfigured = tables != null;
boolean tablesProvided = !tables.isEmpty();
if(tablesConfigured && tablesProvided) {
  // do stuff
} else {
   // try to find them.
}